core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -112,7 +126,12 @@ public void releaseClient(BlockWorkerClientService.Client client) {
   @Override
   public void close() {
     if (mHeartbeat != null) {
-      mHeartbeat.cancel(true);
+      HEARTBEAT_CANCEL_POOL.submit(new Runnable() {
+        @Override
+        public void run() {
+          mHeartbeat.cancel(true);
+        }
+      });
```

Comment:
Yeah, it is intended. It is slow to do interruption between threads. It improves local random read latency with this.